### PR TITLE
Add workflow execution JSON schema

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
Define workflow execution JSON schema, so the users can know the event schema when fetching cloudevent from SNS

Flyte admin will publish a cloudevent with the attribute *jsonSchemaURL* and the value `https://github.com/flyteorg/flyteidl/jsonschema/workflow_execution.json`

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
had already used online JSON schema validator to check the syntax
https://www.jsonschemavalidator.net/

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1019

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by pingsutw** [comment]
We can create flyte cluster by running the below command.
FlyteConsole: `8088`
FlyteAdmin port: `30081`
k8s dashboard: `30082`
```
docker run -it --rm --privileged -p 8088:8088 -p 30081:30081 -p 30082:30082 pingsutw/sandbox-lite
```
Manually build the image
```
docker build -t sandbox-lite -f docker/sandbox-lite/Dockerfile .
```

File: jsonschema/workflow_execution.json [hunk 1]
```diff
@@ -0,0 +1,90 @@
+{
```
Comment (katrogan) [inline jsonschema/workflow_execution.json:1]: can we add comments to this? or a README to the directory perhaps?

Comment (katrogan) [inline jsonschema/workflow_execution.json:1]: can we add comments to this? or a README to the directory perhaps?

Comment (pingsutw) [reply jsonschema/workflow_execution.json:1]: done

File: jsonschema/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# JSON SCHEMA
+
+This folder contains the JSON Schema of execution event, and this schema will
+be used by [cloudevent publisher](https://github.com/flyteorg/flyteadmin/blob/a0ca4b07d3b1c3cccfe3830307df50bc73152ddb/pkg/async/cloudevent/implementations/cloudevent_publisher.go#L96).
+
+The URL for the JSON file will be included with the cloudevent message. For example
+```json
+{
+    "specversion" : "1.0",
+    "type" : "com.flyte.resource.workflow",
```
Comment (katrogan) [inline jsonschema/README.md:10]: what is the corresponding type for node and task execution events?

Comment (pingsutw) [reply jsonschema/README.md:10]: In our last discussion, we decided that we would only support workflow execution events for now, so I only added cloudevent publisher to `execution_manager.go` 
https://github.com/flyteorg/flyteadmin/pull/355/files#diff-9b311b663f43eccf79220bfcae3a565ad26d83532c945f39cfc1f7ba029c22e3R98-R100

**Review by katrogan — APPROVED** [review]

File: jsonschema/README.md (hunk 2 above)
Comment (wild-endeavor) [reply jsonschema/README.md:10]: very java

File: jsonschema/workflow_execution.json [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+          "type": "string"
+        },
+        "phase": {
+          "type": "integer"
+        },
+        "occurred_at": {
+          "$ref": "#/$defs/Timestamp"
+        },
+        "output_result": {
+          "oneOf": [
```
Comment (wild-endeavor) [inline jsonschema/workflow_execution.json:24]: is oneOf a jsonschema thing also?  or just a proto thing?

Comment (pingsutw) [reply jsonschema/workflow_execution.json:24]: yeah, jsonschema also has **oneOf**. https://json-schema.org/understanding-json-schema/reference/combining.html#oneof

**Review by wild-endeavor — APPROVED** [review]
